docs: translate `<textarea>` reference (#498)
reactjs/id.react.dev: src/content/reference/react-dom/components/textarea.md

<Intro>

> komponen [bawaan peramban `<textarea>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) yang memungkinkan Anda *render* teks input dengan banyak baris.

Reviewer: First letter is capital

```suggestion
Komponen [bawaan peramban `<textarea>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) yang memungkinkan Anda me-*render* teks input dengan banyak baris.
```

---

## Referensi {/*reference*/}

### `<textarea>` {/*textarea*/}

> Untuk menampilkan sebuah area teks, *render* [komponen bawaan peramban `<textarea>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea).

Reviewer: ```suggestion
Untuk menampilkan sebuah area teks, me-*render* [komponen `<textarea>` bawaan peramban](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea).
```

```js
<textarea name="postContent" />
```

[Lihat lebih banyak contoh di bawah](#usage)

#### Props {/*props*/}

`<textarea>` mendukung semua [elemen props yang umum.](/reference/react-dom/components/common#props)

> Anda dapat [membuat sebuah area teks yang terkendali](#controlling-a-text-area-with-a-state-variable) dengan cara memberikan sebuah prop `value`:

Reviewer 2: ```suggestion
Anda dapat [membuat sebuah area teks yang terkendali](#controlling-a-text-area-with-a-state-variable) dengan cara memberikan sebuah *prop* `value`:
```

> Anda dapat [membuat sebuah area teks yang terkendali](#controlling-a-text-area-with-a-state-variable) dengan cara memberikan sebuah *prop* `value`:

Reviewer: ```suggestion
Anda dapat [membuat sebuah area teks yang terkendali (*controlled*)](#controlling-a-text-area-with-a-state-variable) dengan cara mengoper sebuah *prop* `value`:
```

* `value`: Sebuah string. Mengontrol teks di dalam area teks.

> Ketika Anda memberikan `value`, Kamu harus memberikan juga sebuah `onChange` *handler* yang memperbarui nilai yang diberikan sebelumnya.

Reviewer: Be consistent, use `Anda`, not `Kamu`.

```suggestion
Ketika Anda memberikan `value`, Anda harus memberikan juga sebuah *handler* `onChange` yang memperbarui nilai yang diberikan sebelumnya.
```

Jika `<textarea>` Anda tidak terkendali, Anda boleh memberikan `defaultValue` sebagai gantinya:

* `defaultValue`: Sebuah string. Menentukan [nilai awal](#providing-an-initial-value-for-a-text-area) untuk sebuah area teks.

> `<textarea>` props ini relevan baik untuk area text terkendali maupun tidak terkendali:

Reviewer 2: ```suggestion
`<textarea>` *props* ini relevan baik untuk area text terkendali maupun tidak terkendali:
```

* [`autoComplete`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-autocomplete): Nilainya `'on'` atau `'off'`. Menentukan perilaku penyelesaian otomatis.
* [`autoFocus`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-autofocus): Sebuah boolean. Jika `true`, React akan memfokuskan elemen ketika terpasang.

> * `children`: `<textarea>` tidak menerima *children*. Untuk Menentukan nilai awal, gunakan `defaultValue`.

Reviewer 2: ```suggestion
* `children`: `<textarea>` tidak menerima anak (*children*). Untuk Menentukan nilai awal, gunakan `defaultValue`.
```

Mungkin tidak familiar dengan anak namun kita mencoba untuk menerapkan perubahann tersebut (Untuk menghindari kehilangan makna, maka ditambahkan english setelah kata seperti anak ini)

> * `children`: `<textarea>` tidak menerima anak (*children*). Untuk Menentukan nilai awal, gunakan `defaultValue`.

Reviewer: Lowercase "menentukan"

```suggestion
* `children`: `<textarea>` tidak menerima anak (*children*). Untuk menentukan nilai awal, gunakan `defaultValue`.
```

> * [`cols`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-cols): Sebuah bilangan. Menentukan lebar bawaaan pada rata-rata lebar karakter. Nilai bawaan adalah `20`.

Reviewer: ```suggestion
* [`cols`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-cols): Sebuah angka. Menentukan lebar bawaaan pada rata-rata lebar karakter. Nilai bawaan adalah `20`.
```

> * [`disabled`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-disabled): Sebuah boolean. Jika `true`, input tidak akan menjadi interaktif dan akan terlihat redup.

Reviewer: input = masukan
kalau mau stay input, pake italic.

```suggestion
* [`disabled`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-disabled): Sebuah boolean. Jika `true`, masukan tidak akan menjadi interaktif dan akan terlihat redup.
```

> * [`form`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-form): Sebuah string. Menentukan `id` pada suatu `<form>` yang memiliki input tersebut. Jika dihilangkan, nilainya mengacu pada induk formulir terdekat.

Reviewer: same about `input`

> * [`maxLength`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-maxlength): Sebuah bilangan. Menentukan panjang maksimum teks.

Reviewer: Number = Angka.

> * [`minLength`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-minlength): Sebuah bilangan. Menentukan panjang minimum teks.

Reviewer: same about `number`

> * [`name`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name): Sebuah string. Menentukan nama pada input yang [dikirim dengan formulir tertentu.](#reading-the-textarea-value-when-submitting-a-form)

Reviewer: same about `input`

* `onSelectCapture`: Sebuah versi `onSelect` yang beroperasi pada [fase penangkapan.](/learn/responding-to-events#capture-phase-events)
* [`placeholder`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-placeholder): Sebuah string. Ditampilkan dalam warna redup ketika nilai area teks kosong.
* [`readOnly`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-readonly): Sebuah boolean. Jika `true`, area teks tidak dapat diubah oleh pengguna.
* [`required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-required): Sebuah boolean. Jika `true`, nilai harus disediakan agar formulir dapat terkirim.
* [`rows`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-rows): Sebuah bilangan. Menentukan tinggi bawaaan pada rata-rata tinggi karakter. Nilai bawaaan adalah `2`.
* [`wrap`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#attr-wrap): Nilainya `'hard'`, `'soft'`, atau `'off'`. Menentukan bagaimana suatu teks akan dibungkus ketika mengirimkan formulir.

#### Caveats {/*caveats*/}

> - Memberikan *children* seperti `<textarea>something</textarea>` tidak diperbolehkan. [Gunakan `defaultValue` untuk konten awal.](#providing-an-initial-value-for-a-text-area)

Reviewer 2: Idem

> - Memberikan anak (*children*) seperti `<textarea>something</textarea>` tidak diperbolehkan. [Gunakan `defaultValue` untuk konten awal.](#providing-an-initial-value-for-a-text-area)

Reviewer: Ketinggalan ini `passing`

- Jika menerima sebuah *prop* `value` string, sebuah area teks akan [dianggap sebagai komponen terkendali (*controlled*).](#controlling-a-text-area-with-a-state-variable)
- Sebuah area teks tidak dapat menjadi terkendali (*controlled*) dan tidak terkendali (*uncontrolled*) secara bersamaan.
- Sebuah area teks tidak dapat beralih menjadi terkendali (*controlled*) atau tidak terkendali (*uncontrolled*) selama masa pakainya.
- Setiap area teks terkendali (*controlled*) membutuhkan sebuah *event handler* `onChange` yang memperbarui nilai pendukungnya secara sinkron.

---

## Penggunaan {/*usage*/}

### Menampilkan sebuah area teks {/*displaying-a-text-area*/}

> *Render* `<textarea>` untuk menampilkan sebuah area teks. Kamu dapat menentukan ukuran bawaanya dengan atribut [`rows`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#rows) dan [`cols`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#cols), tapi secara bawaan user dapat mengubah ukurannya. Untuk menonaktifkan pengubahan ukuran, kamu dapat menentukan `resize: none` di dalam CSS.

Reviewer 3: ```suggestion
*Render* `<textarea>` untuk menampilkan sebuah area teks. Anda dapat menentukan ukuran bawaanya dengan atribut [`rows`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#rows) dan [`cols`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#cols), tapi secara bawaan user dapat mengubah ukurannya. Untuk menonaktifkan pengubahan ukuran, Anda dapat menentukan `resize: none` di dalam CSS.
```

kamu -> anda

---

### Menyediakan sebuah nilai awal pada sebuah area teks {/*providing-an-initial-value-for-a-text-area*/}

> Kamu dapat menentukan nilai awal pada suatu area teks secara opsional. Untuk mengoper nilai awal, gunakan `defaultValue` dengan tipe string.

Reviewer 3: ```suggestion
Anda dapat menentukan nilai awal pada suatu area teks secara opsional. Untuk mengoper nilai awal, gunakan `defaultValue` dengan tipe string.
```

Idem.

Author: resolved

<Note>

Berikan sebuah `name` kepada `<textarea>` Anda, misalkan `<textarea name="postContent" />`. `name` yang Anda tetapkan akan digunakan sebagai sebuah *key* di dalam data formulir, misalkan `{ postContent: "Your post" }`.

</Note>

<Pitfall>

Secara *default*, *setiap* `<button>` yang berada di dalam sebuah `<form>` akan mengirimkan data formulir tersebut. Ini bisa mengejutkan! Jika Anda mempunyai kustom komponen React `Button` Anda sendiri, pertimbangkanlah untuk mengembalikan [`<button type="button">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/button) daripada `<button>`. Kemudian, supaya menjadi eksplisit, gunakan `<button type="submit">` untuk tombol-tombol yang seharusnya digunakan untuk mengirimkan formulir.

</Pitfall>

---

### Mengendalikan sebuah area teks dengan sebuah variabel state {/*controlling-a-text-area-with-a-state-variable*/}

Sebuah area teks seperti `<textarea />` bersifat *tak terkendali (uncontrolled).* Meskipun jika Anda [mengoper sebuah nilai awal](#providing-an-initial-value-for-a-text-area) seperti `<textarea defaultValue="Initial text" />`, JSX Anda hanya menetapkan nilai awal, bukan nilai saat ini.

> **Untuk me-*render* sebuah teks area _terkendali_, berikan *prop* `value` kepada area teksnya.** React akan memaksa area teks tersebut agar selalu mempunyai `value` yang Anda berikan. Umumnya, kamu akan mengendalikan sebuah area teks dengan mendeklarasikan sebuah [variabel *state*:](/reference/react/useState)

Reviewer 3: ```suggestion
**Untuk me-*render* sebuah teks area _terkendali_, berikan *prop* `value` kepada area teksnya.** React akan memaksa area teks tersebut agar selalu mempunyai `value` yang Anda berikan. Umumnya, Anda akan mengendalikan sebuah area teks dengan mendeklarasikan sebuah [variabel *state*:](/reference/react/useState)
```

idem.

Author: resolved

> **Untuk me-*render* sebuah teks area _terkendali_, berikan *prop* `value` kepada area teksnya.** React akan memaksa area teks tersebut agar selalu mempunyai `value` yang Anda berikan. Umumnya, kamu akan mengendalikan sebuah area teks dengan mendeklarasikan sebuah [variabel *state*:](/reference/react/useState)

Reviewer 3: ```suggestion
**Untuk me-*render* sebuah teks area _terkendali_, berikan *prop* `value` kepada area teksnya.** React akan memaksa area teks tersebut agar selalu mempunyai `value` yang Anda berikan. Umumnya, Anda akan mengendalikan sebuah area teks dengan mendeklarasikan sebuah [variabel *state*:](/reference/react/useState)
```

idem.

```js
// ✅ Good: area teks terkontrol yang hanya untuk dibaca tanpa onChange
<textarea value={something} readOnly={true} />
```

---

### Tanda sisipan saya melompat ke awal pada setiap penekanan tombol {/*my-text-area-caret-jumps-to-the-beginning-on-every-keystroke*/}

Jika Anda [mengendalikan sebuah area teks,](#controlling-a-text-area-with-a-state-variable) Anda harus memperbarui variabel *state*-nya ke nilai area teksnya melalui DOM selama `onChange`.

Anda tidak dapat memperbarui nilainya dengan sesuatu selain `e.target.value`:

```js
function handleChange(e) {
  // 🔴 Bug: memperbarui sebuah masukan dengan sesuatu selain e.target.value
  setFirstName(e.target.value.toUpperCase());
}
```

Anda juga tidak dapat memperbaruinya secara asinkron:

```js
function handleChange(e) {
  // 🔴 Bug: memperbarui sebuah masukan secara asinkron
  setTimeout(() => {
    setFirstName(e.target.value);
  }, 100);
}
```

Untuk memperbaiki kode Anda, perbarui secara sinkron ke `e.target.value`:

```js
function handleChange(e) {
  // ✅ Memperbarui sebuah masukan terkendali ke e.target.value secara sinkron
  setFirstName(e.target.value);
}
```

> Jika ini tidak menyelesaikan masalah, ada kemungkinan area teks terhapus dan ditambahkan kembali dari DOM di setiap penekanan tombol. Ini dapat terjadi jika Anda secara tidak sengaja [menyetel ulang *state*](/learn/preserving-and-resetting-state) di setiap *render* ulang. Misalkan, ini dapat terjadi jika area teksnya atau salah satu dari *parents*-nya selalu menerima sebuah atribut `key` yang berbeda, atau jika Anda menyusun definisi komponen (dimana hal ini tidak diizinkan React dan menyebabkan komponen "dalam" terpasang ulang di setiap *render*).

Reviewer: Kalau mount diterjemahkan "pasang" sepertinya kurang pas.
How about keep as it is. So it will be remount = `ter-*mount* ulang`

Author: yeah, agree with you. updated

---

> ### Saya menerima sebuah error: "Sebuah komponen sedang mengubah masukan yang tidak terkendali (*uncontrolled*) menjadi terkendali (*controlled*)" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}

Reviewer 3: ```suggestion
### Saya menerima sebuah error: "Sebuah komponen sedang mengubah masukan yang tidak terkendali (uncontrolled) menjadi terkendali (controlled)" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}
```

Don't italicise loanwords in headings.

https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Umum#kata-serapan-di-dalam-heading

> ### Saya menerima sebuah error: "Sebuah komponen sedang mengubah masukan yang tidak terkendali (*uncontrolled*) menjadi terkendali (*controlled*)" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}

Reviewer 3: ```suggestion
### Saya menerima sebuah error: "Sebuah komponen sedang mengubah masukan yang tidak terkendali (uncontrolled) menjadi terkendali (controlled)" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}
```

Don't italicise loanwords in headings.

https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Umum#kata-serapan-di-dalam-heading

> ### Saya menerima sebuah error: "Sebuah komponen sedang mengubah masukan yang tidak terkendali (*uncontrolled*) menjadi terkendali (*controlled*)" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}

Reviewer 3: ```suggestion
### Saya menerima sebuah error: "Sebuah komponen sedang mengubah masukan yang tidak terkendali (uncontrolled) menjadi terkendali (controlled)" {/*im-getting-an-error-a-component-is-changing-an-uncontrolled-input-to-be-controlled*/}
```

Don't italicise loanwords in headings.

https://github.com/reactjs/id.react.dev/wiki/Panduan-Penulisan-Umum#kata-serapan-di-dalam-heading
